### DISCUSSION TITLE
ZSH as default shell

### DISCUSSION THREAD

**Discussion by surendrajat** [elementary/terminal#555] [Ideas]
Guys, what you think of this idea of shipping ZSH as the default shell instead of Bash? Having a curated theme preinstalled will also be super cool to see.
It is the default shell now for Kali Linux, Garuda Linux, etc., and has been set as default for macOS for a long time now as well.

**Comment by danirabbit** [comment]
@Surendrajat Can you provide some details about the benefits of ZSH over BASH? What specific features does it provide or problems does it solve? Are there any drawbacks?

**Comment by surendrajat** [comment]
@danrabbit Here are some features of `ZSH` which I'm aware of:
- **Autosuggestion/completion**: super useful. You know how you browse the history using UP arrow in bash. In ZSH, you can do that in middle of typing and relative to why you typed(as in the ss). This feature alone was enough for me to set it as defualt. Also, when you press TAB for completion it doesn't just spit available options but you can also cycle though them using TAB.
- **Syntax highlighting**: See ss.
- **Plugins & Themes**: with frameworks like [oh-my-zsh](https://github.com/ohmyzsh/ohmyzsh) and themes like [powerlevel10k](https://github.com/romkatv/powerlevel10k) you can truely personalise it to your taste and for a better workflow.
- **Useful prompt**: what if you never typed `git status` ever again and you still know the `git status`? :) And it adjusts itself when things get too long for a prompt.
- More: like **Auto-correction/capitalization** of things you type (useful for dir names, etc.), no need to type `cd` if you dont want to, and stuff.
![zsh](https://user-images.githubusercontent.com/11215000/103398814-474d6780-4b64-11eb-8518-55291b7fff56.gif)

Now, I know that `BASH` is the SHELL for most users and it just works. Further, there are efforts to make it more useful with things like oh-my-bash & git-prompt-bash.
But ZSH gives you all that + all the cool stuff(matured) with good visual apeal and still feels perfoment. With right set of plugins and config you can do the same job without even half of keystokes you'd do in BASH. As if the SHELL almost understands you :) As a DevOps Engineer, I appriciate it a lot and hope Ubuntu(servers we use) also sets it as default with some basic set of plugins, at least.

It is mostly compitible with Bash behaviour and when its not [it can be configured to do so](https://unix.stackexchange.com/questions/130958/scp-wildcard-not-working-in-zsh). All bash scripts will continue to run as long as bash is installed in system.

As for the drawbacks, I don't see any except, of course, the obivious one: Customizability. But I think with proper communication and an easy to switch guide, this shouldn't be of much concern and users living in terminal will mostly appreciate it.

**Comment by parnoldx** [comment]
Also to consider. The Terminal tab title feature uses the bashrc to work. If changed to zsh this has to ported to whatever zsh offers.
Here is the code https://bazaar.launchpad.net/~elementary-os/elementaryos/os-patch-bash-bionic/view/head:/debian/skel.bashrc#L114
The https://github.com/rcaloras/bash-preexec which this feature uses seems to mimic zsh so this should be fine.

**Comment by Kissanth** [comment]
Why not we can put the option for user can choose our preference in terminal preference section

**Reply by Kissanth** [reply]
And also you can control how many plugin are used in terminal it's a optional but not sure how that's should be work
